tests/test_multiannotator.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pytest
+from cleanlab import dataset
+from cleanlab.internal.label_quality_utils import _subtract_confident_thresholds
+from cleanlab.benchmarking.noise_generation import generate_noise_matrix_from_trace
+from cleanlab.benchmarking.noise_generation import generate_noisy_labels
+from cleanlab import count
+from cleanlab.multiannotator import get_label_quality_scores_multiannotator
+import pandas as pd
+
+
+def make_data(
+    means=[[3, 2], [7, 7], [0, 8]],
+    covs=[[[5, -1.5], [-1.5, 1]], [[1, 0.5], [0.5, 4]], [[5, 1], [1, 5]]],
+    sizes=[80, 40, 40],
+    avg_trace=0.8,
+    num_annotators=5,
+    seed=1,  # set to None for non-reproducible randomness
+):
+    np.random.seed(seed=seed)
+
+    m = len(means)  # number of classes
+    n = sum(sizes)
+    local_data = []
+    labels = []
+    test_data = []
+    test_labels = []
+
+    for idx in range(m):
+        local_data.append(
+            np.random.multivariate_normal(mean=means[idx], cov=covs[idx], size=sizes[idx])
+        )
+        test_data.append(
+            np.random.multivariate_normal(mean=means[idx], cov=covs[idx], size=sizes[idx])
+        )
+        labels.append(np.array([idx for i in range(sizes[idx])]))
+        test_labels.append(np.array([idx for i in range(sizes[idx])]))
+    X_train = np.vstack(local_data)
+    true_labels_train = np.hstack(labels)
+    X_test = np.vstack(test_data)
+    true_labels_test = np.hstack(test_labels)
+
+    # Compute p(true_label=k)
+    py = np.bincount(true_labels_train) / float(len(true_labels_train))
+
+    noise_matrix = generate_noise_matrix_from_trace(
+        m,
+        trace=avg_trace * m,
+        py=py,
+        valid_noise_matrix=True,
+        seed=seed,
+    )
+
+    # Generate our noisy labels using the noise_matrix for specified number of annotators.
+    s = pd.DataFrame(
+        np.vstack(
+            [generate_noisy_labels(true_labels_train, noise_matrix) for _ in range(num_annotators)]
+        ).transpose()
+    )
+
+    # Each annotator only labels approximately 75% of the dataset
```

Comment:
maybe we want to test with more sparsity? Eg. each annotator only labels say ~20%?

Reply:
If more sparsity is utilized, probably need to delete rows that have no annotators (or keep at least 1 annotator for such rows)

Reply:
TODO: edge case here that will cause `get_multiannotator_stats` to fail - `dataset.overall_label_health_score` requires that there are no missing labels, which is not guaranteed here (some annotators might label too few examples, or bad annotator might label same labels for all examples)